### PR TITLE
♻️ Refactor links in target api cfg 

### LINKED ISSUE #318: Explicitly define target concepts in target api config links
Reformat target API links so that the target concept being linked is more explicit. The lib currently pulls the target concept from the link attribute name. 

This is bad and doesn't allow us to support a target concept config that may have two or more links to the same target concept (e.g. family relationship has links to two of the same target concept, participant) 

Change this:
```
'participant': {
        'standard_concept': CONCEPT.PARTICIPANT,
        'links': {
            'family_id': CONCEPT.FAMILY.UNIQUE_KEY,
            'study_id': CONCEPT.STUDY.UNIQUE_KEY
        },
       ...
    },
```
to this
```
'participant': {
        'standard_concept': CONCEPT.PARTICIPANT,
        'links': [ 
            {
                'standard_concept': CONCEPT.FAMILY.UNIQUE_KEY, 
                'name': 'family_id' ,
                'target_concept': 'family'
            },
            {
                'standard_concept': CONCEPT.STUDY.UNIQUE_KEY,
                'name': study_id',
                'target_concept': 'study'
            }
        ],
    },
```

### PR DESCRIPTION
**Pull request by znatty22**
Closes #318

### CODE REVIEW

File: kf_lib_data_ingest/etl/configuration/target_api_config.py [hunk 1]
```diff
@@ -227,52 +234,98 @@ def _validate_target_concept_attr_mappings(self):
...
+            sc = link_dict['standard_concept']
+            if sc not in concept_set:
+                sc_name = sc
+                try:
+                    sc_name = sc._CONCEPT_NAME
+                except AttributeError:
+                    pass
+                raise ValueError(
+                    'Invalid link found in target concept config \n'
+                    f'{pformat(target_concept_dict)}! '
```
Comment (fiendish) [inline kf_lib_data_ingest/etl/configuration/target_api_config.py]: It's probably not a good idea to put punctuation adjacent to a data structure.

```suggestion
                    f'{pformat(target_concept_dict)}'
```

File: kf_lib_data_ingest/etl/configuration/target_api_config.py [hunk 2]
```diff
@@ -227,52 +234,98 @@ def _validate_target_concept_attr_mappings(self):
...
+                    f'\nThe linked standard concept: {sc_name} does '
+                    'not exist in the standard concept set!'
+                )
+            # Linked target concepts must be one of the target concept
+            # keys
+            tc = link_dict['target_concept']
+            if tc not in valid_target_concepts:
+                raise ValueError(
+                    'Invalid link found in target concept config '
+                    f'{pformat(target_concept_dict)}! The value of a '
```
Comment (fiendish) [inline kf_lib_data_ingest/etl/configuration/target_api_config.py]:
```suggestion
                    f'{pformat(target_concept_dict)}\nThe value of a '
```

File: kf_lib_data_ingest/etl/configuration/target_api_config.py [hunk 3]
```diff
@@ -227,52 +234,98 @@ def _validate_target_concept_attr_mappings(self):
...
+                    'linked target concept must point to one of the '
+                    'existing target concepts: '
+                    f'{pformat(valid_target_concepts)}\n Target '
+                    f'concept {tc} does not exist'
```
Comment (fiendish) [inline kf_lib_data_ingest/etl/configuration/target_api_config.py]:
```suggestion
                    f'concept {tc} does not exist.'
```

File: kf_lib_data_ingest/etl/configuration/target_api_config.py [hunk 4]
```diff
@@ -58,16 +58,21 @@
...
+                        - target_attribute: a string containing the target
+                        concept property
+                        - standard concept: a standard concept from
+                        etl.transform.standard_model.concept_schema
+                        - target_concept: name of the linked target concept
+
+                    dict example:
+                    {
+                        'target_attribute': 'family_id',
+                        'standard_concept': CONCEPT.FAMILY,
```
[the remark this answers is not in the document]
Comment (znatty22) [reply kf_lib_data_ingest/etl/configuration/target_api_config.py]: No its supposed to point to the concept. The `standard_concept` in links follows the same format as the `standard_concept` in the top level of the target concept's schema

File: kf_lib_data_ingest/etl/configuration/target_api_config.py [hunk 5]
```diff
@@ -84,20 +89,21 @@
...
+        and then kept as is during the load stage.
 
-        For example, after transformation and before loading, the 'links' dict
+        For example, before ID translation in load stage, the 'links' list
         could be:
 
-            'links': {
+            'links': [{
+                'target_concept': 'participant',
                 'participant_id': 'CONCEPT|PARTICIPANT|UNIQUE_KEY|P1'
```
[the remark this answers is not in the document]
Comment (znatty22) [reply kf_lib_data_ingest/etl/configuration/target_api_config.py]: Oh I think that is just supposed to be P1

**Pull request merged**